ember-resources/src/util/function.ts

```diff
@@ -272,6 +272,12 @@ export class State<Value> {
   retry = async () => {
     if (isDestroyed(this) || isDestroying(this)) return;
 
+    // We've previously had data, but we're about to run-again.
+    // we need to do this again so `isLoading` goes back to `true` when re-running.
+    if (this.data) {
+      this.data = null;
```

Comment:
this is the fix

Reply:
This makes a lot of sense. And these code comments are great